web/client.js

```diff
@@ -871,18 +871,52 @@ function sendGlobalMessage() {
         vm.globalMessage('');
     }
 }
-
-var windowVisible = true;
-$(window).on('focus', function () {
-    windowVisible = true;
-});
-$(window).on('blur', function () {
-    windowVisible = false;
-});
-function notifyPlayer(message) {
-    if (vm.notifsEnabled() && !windowVisible) {
+function registerServiceWorker() {
+    if ('serviceWorker' in navigator) {
+        window.addEventListener('load', () => {
+            navigator.serviceWorker.register('/service-worker.js').then((registration) => {
+                console.log('ServiceWorker registration successful with scope: ', registration.scope);
+            }, (err) => {
+                console.log('ServiceWorker registration failed: ', err);
+            });
+        });
+    }
+}
+function listenForNotificationAction() {
+    window?.navigator?.serviceWorker?.addEventListener('message', event => {
```

Comment:
`?.` is not supported in Firefox for Android (quite a few people use this) and some other mobile browsers. I try to keep the code as compatible as possible, so would you mind removing these? Most of these references cannot be null anyway (like `window` and `navigator`).

It was only recently that the first arrow function made its way into the code, so goodbye Internet Explorer I guess (and good riddance!)